azure/scripts/ansible/playbooks/roles/autoscaler/tasks/main.yml

```diff
@@ -3,7 +3,7 @@
   ansible.builtin.command: |
     oc get machineset -n openshift-machine-api -o jsonpath='{.items[0].metadata.labels.machine\.openshift\.io/cluster-api-cluster}'
   register: CLUSTER_ID_OUTPUT
-  changed_when: CLUSTER_ID_OUTPUT.rc != 0
+  changed_when: CLUSTER_ID_OUTPUT.rc == 0
```

Comment:
I still think this is improper usage of the `changed_when` logic. 

As per the [doc](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_error_handling.html#defining-changed), the `changed_when` option should be used to define additional criteria for what would be considered a successful change to a remote system.

In the `command` you're invoking above this, there's no changes that happen on the remote machine by doing an `oc get`. When you look up the ansible lint rule, the first hit on Google is this [issue](https://github.com/geerlingguy/ansible-role-certbot/issues/144), which tells you to just use `changed_when: false`... but even that seems non-ideal.

What's important to realize is if you create a sample playbook with only the task above and run it with the `changed_when` condition commented out... ansible will flag that command as a changed just because the RC is 0... so `changed_when: rc == 0` is redundant to the internal workings of Ansible... and the only benefit that addition has is to remove the ansible lint failure.